sdk/daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/preprocessing/Preprocessor.scala

```diff
@@ -202,6 +209,27 @@ private[engine] final class Preprocessor(
       } yield m.updated(pkgName, pkgId)
     )
 
+  def buildGlobalKey(
+      templateId: Ref.TypeConName,
```

Comment:
Do you need this only for a template ID or also for template references?

Reply:
Currently the resolution from package-name to package-id is done on the Canton side. The API is slightly different from `Engine.submit` but in this case:
- Only a single entry package is involved (`submit` may have several commands)
- Package preference is not a consideration
- The default logic on the canton side is to use the latest package version where a package name is provided.